### DISCUSSION TITLE
history.push跳转不了？

### DISCUSSION THREAD

**Discussion by andyFang94** [umijs/umi#12256] [Q&A]
各位有没有遇到过，点击按钮执行   

```
import { history } from '@umijs/max';

 history.push( `/page/${id}?name=${name}` );
```

页面跳转不了？但是浏览器地址栏已经变了，页面还没发生跳转， 需要过一会才能跳转过来（偶尔发生）

**Comment by fz6m** [comment]
有一种可能是你的逻辑没优化好，或者逻辑太多了，或者重渲染涉及的组件太多了，或者下一个页面的组件渲染太费时了等等，这种情况就会卡一下然后跳转过去。

得找一个可以必现的例子，看看是不是因为某处逻辑太浪费时间了或者监听器太多了等逻辑上的原因导致的，或者你把控制台关了试试，对比一下不关控制台两者间的速度，如果关了控制台就变快了，这种是正常情况，因为开着控制台浏览器会跟踪调用栈，如果逻辑多了，做操作会卡是正常的，不仅仅是 history 跳转。

如还有问题，需要给一个最小复现。
